src/verifiable-credential.js

```diff
@@ -178,12 +178,7 @@ class VerifiableCredential {
    */
   addSubject(subject) {
     ensureObjectWithId(subject, 'credentialSubject');
-    if (!this.credentialSubject || this.credentialSubject.length === 0) {
-      this.credentialSubject = [subject];
-    }
-
-    const subjects = this.credentialSubject.length ? this.credentialSubject : [this.credentialSubject];
-    this.credentialSubject = getUniqueElementsFromArray([...subjects, subject], JSON.stringify);
+    this.credentialSubject = getUniqueElementsFromArray([...this.credentialSubject, subject], JSON.stringify);
```

Comment:
was this a bad merge? the code removed fixes bugs with subject undefined/as object (normally happens with fromjson method creation)

Reply:
Actually if I remember correctly we ran a test and the deleted line 185 was causing a bug where adding a subject resulted in a two-array subject, one of which was an empty array.

Reply:
line 185 does cause that bug, but the code i added above that that is also deleted here fixed it, this has reverted to before it was fixed

Reply:
similar fix was implemented for this feedback https://github.com/docknetwork/sdk/pull/110#discussion_r428340774